frontend/packages/employee-frontend/src/components/messages/MessageList.tsx

```diff
@@ -2,85 +2,171 @@
-                  </MessageTitle>
-                  {msg.sentAt ? (
-                    <FixedSpaceRow>
-                      <span>
-                        {groups.find((g) => g.id === msg.groupId)?.name}
-                      </span>
-                      <span>
-                        {formatDate(
-                          msg.sentAt,
-                          LocalDate.fromSystemTzDate(msg.sentAt).isEqual(
-                            LocalDate.today()
-                          )
-                            ? 'HH:mm'
-                            : 'd.M.'
-                        )}
-                      </span>
-                    </FixedSpaceRow>
-                  ) : (
-                    <span>{i18n.messages.notSent}</span>
-                  )}
-                </MessageTopRow>
-                <MessageSummary>
-                  {msg.content.substring(0, 200).replace('\n', ' ')}
-                </MessageSummary>
-              </MessageListItem>
-              <StyledHr />
-            </React.Fragment>
-          ))}
-        </>
+      {messagesState.bulletins.map((msg) => (
+        <React.Fragment key={msg.id}>
+          <MessageListItem onClick={() => selectMessage(msg)}>
+            <MessageTopRow>
+              <MessageTitle noMargin>
+                {msg.title || `(${i18n.messages.noTitle})`}
+              </MessageTitle>
+              {msg.sentAt ? (
+                <FixedSpaceRow>
+                  <span>{groups.find((g) => g.id === msg.groupId)?.name}</span>
+                  <span>
+                    {formatDate(
+                      msg.sentAt,
+                      LocalDate.fromSystemTzDate(msg.sentAt).isEqual(
+                        LocalDate.today()
+                      )
+                        ? 'HH:mm'
+                        : 'd.M.'
+                    )}
+                  </span>
+                </FixedSpaceRow>
+              ) : (
+                <span>{i18n.messages.notSent}</span>
+              )}
+            </MessageTopRow>
+            <MessageSummary>
+              {msg.content.substring(0, 200).replace('\n', ' ')}
+            </MessageSummary>
+          </MessageListItem>
+          <StyledHr />
+        </React.Fragment>
+      ))}
+      <OnEnterView onEnter={loadNextPage} />
+      {messagesState.nextPage.isLoading && <SpinnerSegment />}
+      {messagesState.nextPage.isFailure && (
+        <ErrorSegment title={i18n.common.loadingFailed} />
       )}
     </Container>
   )
 })
 
+interface MessagesState {
+  bulletins: Bulletin[]
+  nextPage: Result<void>
+  currentPage: number
+  pages: number
+  total?: number
+}
+
+const initialState: MessagesState = {
+  bulletins: [],
+  nextPage: Loading.of(),
+  currentPage: 1,
+  pages: 0
+}
+
+const OnEnterView = React.memo(function IsInView({
+  onEnter
+}: {
+  onEnter: () => void
+}) {
+  const ref = useIntersectionObserver<HTMLDivElement>(onEnter)
+  return <div ref={ref} />
+})
```

Comment:
![Screenshot 2021-03-04 at 11 39 35](https://user-images.githubusercontent.com/1533854/109944034-773cf680-7cde-11eb-9a7c-7825c6ca0da9.png)
Sometimes this does not quite seem to work, there is hardly any scrollbar or it may be completely missing. Maybe adding some height for this div could help or maybe page size 15 is just too small for 1440p screen. Or maybe you could try to have it load next page when this element is visible (now it seems to trigger only when it becomes visible after initially being hidden).

Reply:
Whoops, I was just testing it with a page size of 15, it should have been 50 😅